src/pykiso/lib/connectors/cc_serial.py

```diff
@@ -0,0 +1,181 @@
+        :param xonxoff: Enable software flow contro, defaults to False
+        :param rtscts: Enable hardware (RTS/CTS) flow control, defaults to False
+        :param write_timeout: _description_, defaults to None
+        :param dsrdtr: Enable hardware (DSR/DTR) flow control, defaults to False
+        :param inter_byte_timeout:  Inter-character timeout, None to disable,
+          defaults to None
+        :param exclusive: Set exclusive access mode (POSIX only).
+          A port cannot be opened in exclusive access mode if it is already open
+          in exclusive access mode., defaults to None
+        """
+
+        # Initialize the super class
+        super().__init__(**kwargs)
+        # Initialize the serial connection. Port None prevents automatic opening
+        self.serial = serial.Serial(
+            port=None,
+            baudrate=baudrate,
+            bytesize=bytesize,
+            parity=parity,
+            stopbits=stopbits,
+            timeout=timeout,
+            xonxoff=xonxoff,
+            rtscts=rtscts,
+            write_timeout=write_timeout,
+            dsrdtr=dsrdtr,
+            inter_byte_timeout=inter_byte_timeout,
+            exclusive=exclusive,
+        )
+
+        self.serial.port = port
+
+    def _cc_open(self) -> None:
+        """Open serial port"""
+        self.serial.open()
+
+    def _cc_close(self) -> None:
+        """Close serial port"""
+        self.serial.close()
+
+    def _cc_send(
+        self, msg: Union[bytes, bytearray, memoryview], raw=False, timeout=None
+    ) -> None:
+        """Sends data to the serial port
+
+        :param msg: data to send
+        :param raw: unused
+        :param timeout: write timeout in seconds. None sets it to blocking,
+          defaults to None
+        :raises: SerialTimeoutException - In case a write timeout is configured
+          for the port and the time is exceeded.
+        """
+        self.serial.write_timeout = timeout
+        self.serial.write(msg)
+
+    def _cc_receive(self, timeout=0.00001, raw: bool = False) -> bytes:
+        """Read bytes from the serial port.
+        Try to read one byte in blocking mode. After blocking read check
+        remaining bytes and read them without a blocking call.
+
+        :param timeout: timeout in seconds, defaults to 0.00001
+        :param raw: not implemented, defaults to False
+        :raises NotImplementedError: if raw is to True
+        :return: received bytes
+        """
+        if raw:
+            raise NotImplementedError()
+
+        self.serial.timeout = timeout
+
+        received = self.serial.read()
+
+        # We have waited already so set the timeout to non blocking
+        self.serial.timeout = 0
+
+        # Check how many bytes are still left and read them at once
+        in_waiting = self.serial.in_waiting
+        if in_waiting > 0:
+            received += self.serial.read(in_waiting)
+
+        return received
+
+    def cc_readline(self, timeout=0):
```

Comment:
Why are this APIs needed?

Reply:
I use it for the Automationboard which makes communication very easy and handy.
Its  also a standard method in pyserial and I think that many other devices also ends their messages whith a newline character

Reply:
This will not work with the proxy, please comply here with the dictionary approach to forward the parameters to the read method directly and handle it there :)

Reply:
added dict approach... remove readline